ksql-engine/src/test/java/io/confluent/ksql/function/udf/geo/GeoDistanceKudfTest.java

```diff
@@ -0,0 +1,58 @@
+package io.confluent.ksql.function.udf.geo;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import io.confluent.ksql.function.KsqlFunctionException;
+import static org.junit.Assert.assertEquals;
+
+public class GeoDistanceKudfTest {
+  private GeoDistanceKudf distanceUdf = new GeoDistanceKudf();
+
+  @Rule
+  public final ExpectedException expectedException = ExpectedException.none();
+
+  /**
+   * First, the happy-path test: compute distance between Palo Alto and London offices.
+   */
+  @Test
+  public void shouldComputeDistanceBetweenLocations() {
+    assertEquals(8634.6528, (double) distanceUdf.evaluate(37.4439, -122.1663, 51.5257, -0.1122),
+        0.5);
+    assertEquals(8634.6528,
+        (double) distanceUdf.evaluate(37.4439, -122.1663, 51.5257, -0.1122, "KM"), 0.5);
+    assertEquals(5365.66, (double) distanceUdf.evaluate(37.4439, -122.1663, 51.5257, -0.1122, "MI"),
+        0.5);
+  }
+
+  @Test
+  public void shouldFailWithTooFewParams() {
+    expectedException.expect(KsqlFunctionException.class);
+    expectedException.expectMessage("GeoDistance function expects either 4 or 5 arguments");
+    assertEquals(8634.6528, (double) distanceUdf.evaluate(37.4439, -122.1663), 0.5);
```

Comment:
here and elsewhere, `assertEquals` doesn't make sense as you are expecting an exception

Reply:
good point! will remove...